promql/functions.go

```diff
@@ -996,6 +996,33 @@ func funcHistogramSum(vals []parser.Value, args parser.Expressions, enh *EvalNod
 	return enh.Out
 }
 
+// === histogram_stdvar(Vector parser.ValueTypeVector) Vector ===
+func funcHistogramStdVar(vals []parser.Value, args parser.Expressions, enh *EvalNodeHelper) Vector {
+	inVec := vals[0].(Vector)
+
+	for _, sample := range inVec {
+		// Skip non-histogram samples.
+		if sample.H == nil {
+			continue
+		}
+		mean := sample.H.Sum / sample.H.Count
+		var variance, cVariance float64
+		it := sample.H.AllBucketIterator()
+		for it.Next() {
+			bucket := it.At()
+			mid := (bucket.Upper + bucket.Lower) / 2
```

Comment:
We probably want something here that fits the exponential layout better than the arithmetic mean. I suspect this could be the geometric mean, but I'm not sure yet. Do you have an idea? For now, let's just add an TODO item here.

Reply:
After some thinking, I'm pretty sure we want the geometric mean here, i.e `math.Sqrt(bucket.Upper * bucket.Lower)`.
An exception is the zero bucket, for which we want 0 as the midpoint.